### PR TITLE
[ButtonBar] Allow customizing the inkColor of the buttons

### LINKED ISSUE #1660: AppBar's button ink is the wrong color in MDCCatalog
MDCCatalog has mostly dark colors with a lighter colored ink. The MDCButtonBarButton in the MDCAppBar seems to still have a dark ink applied, so it is barely visible.

Example:
1. Open MDCatalog > App Bar > Demo
1. Tap and hold on the button "Right"
1. Notice that the ink is barely visible as a darker color behind the text.

### PR DESCRIPTION
**Pull request by AlimoRabbani**
Make inkColor for buttons in MDCButtonBar customizable. Also expose an API on MDCNavigationBar to allow using the custom inkColor in AppBar.

closes #1660

### CODE REVIEW

**Comment by AlimoRabbani** [comment]
I'd be happy to write a test for this, but I'll need pointers on where to start, or maybe similar tests?

File: components/ButtonBar/src/private/MDCAppBarButtonBarBuilder.m [hunk 1]
```diff
@@ -92,7 +92,8 @@ - (UIView *)buttonBar:(MDCButtonBar *)buttonBar
   MDCButtonBarButton *button = [[MDCButtonBarButton alloc] init];
   [button setBackgroundColor:[UIColor clearColor] forState:UIControlStateNormal];
   button.disabledAlpha = kDisabledButtonAlpha;
-
+  button.inkColor = buttonBar.inkColor ?: buttonBar.inkColor;
```
Comment (jverkoey) [inline components/ButtonBar/src/private/MDCAppBarButtonBarBuilder.m]: I think the latter part of this expression is meant to be `button.inkColor`.

File: components/ButtonBar/src/MDCButtonBar.h [hunk 2]
```diff
@@ -111,6 +111,12 @@ IB_DESIGNABLE
  */
 - (CGSize)sizeThatFits:(CGSize)size;
 
+/**
+ The Material inkColor that is used for all buttons in the button bar. Only affects buttons that are
+ created after changing the property.
+ */
+@property(nonatomic, strong, nullable) UIColor *inkColor;
```
Comment (jverkoey) [inline components/ButtonBar/src/MDCButtonBar.h]: Document that if this property is nil then the default button ink color will be used instead.

File: components/NavigationBar/src/MDCNavigationBar.h [hunk 3]
```diff
@@ -106,6 +106,12 @@ IB_DESIGNABLE
  */
 @property (nonatomic, strong, null_resettable) UIFont *titleFont;
 
+/**
+ The Material inkColor that is used for all buttons in trailing and leading button bars. Only
+ affects buttons that are created after changing the property.
```
Comment (jverkoey) [inline components/NavigationBar/src/MDCNavigationBar.h]: Document that if this property is nil then the default button ink color will be used instead.

**Review by jverkoey — CHANGES_REQUESTED** [review]
Minor feedback.

File: components/ButtonBar/src/MDCButtonBar.h (hunk 2 above)
Comment (AlimoRabbani) [reply components/ButtonBar/src/MDCButtonBar.h]: Done.

File: components/NavigationBar/src/MDCNavigationBar.h (hunk 3 above)
Comment (AlimoRabbani) [reply components/NavigationBar/src/MDCNavigationBar.h]: Done.

File: components/ButtonBar/src/private/MDCAppBarButtonBarBuilder.m (hunk 1 above)
Comment (AlimoRabbani) [reply components/ButtonBar/src/private/MDCAppBarButtonBarBuilder.m]: Done.

**Review by AlimoRabbani — COMMENTED** [review]
Thanks for the review!

**Comment by AlimoRabbani** [comment]
I think this should be ready.

File: components/AppBar/examples/AppBarTypicalUseExample.m [hunk 4]
```diff
@@ -43,7 +43,7 @@ - (id)init {
     [_appBar.headerViewController.headerView hideViewWhenShifted:_appBar.headerStackView];
 
     _appBar.navigationBar.tintColor = [UIColor whiteColor];
-    _appBar.navigationBar.inkColor = [UIColor colorWithWhite:0.9 alpha:0.1];
+//    _appBar.navigationBar.inkColor = [UIColor colorWithWhite:0.9 alpha:0.1];
```
Comment (jverkoey) [inline components/AppBar/examples/AppBarTypicalUseExample.m]: Revert

**Review by jverkoey — CHANGES_REQUESTED** [review]

File: components/ButtonBar/src/MDCButtonBar.h [hunk 5]
```diff
@@ -105,6 +105,14 @@ IB_DESIGNABLE
  */
 @property(nonatomic) MDCButtonBarLayoutPosition layoutPosition;
 
+/**
+ The Material inkColor that is used for all buttons in the button bar. Only affects buttons that are
```
Comment (jverkoey) [inline components/ButtonBar/src/MDCButtonBar.h]: Remove “Material”.

File: components/NavigationBar/src/MDCNavigationBar.h [hunk 6]
```diff
@@ -113,6 +113,14 @@ IB_DESIGNABLE
  */
 @property(nonatomic, strong, nullable) UIColor *titleTextColor;
 
+/**
+ The Material inkColor that is used for all buttons in trailing and leading button bars. Only
```
Comment (jverkoey) [inline components/NavigationBar/src/MDCNavigationBar.h]: Remove “Material”.

File: components/AppBar/examples/AppBarTypicalUseExample.m (hunk 4 above)
Comment (AlimoRabbani) [reply components/AppBar/examples/AppBarTypicalUseExample.m]: I did in 6b425f5

I might have messed up the review history :(

File: components/ButtonBar/src/MDCButtonBar.h [hunk 7]
```diff
@@ -105,6 +105,14 @@ IB_DESIGNABLE
...
+ created after changing the property.
```
Comment (jverkoey) [inline components/ButtonBar/src/MDCButtonBar.h]: I think that clients will expect this property to affect existing buttons as well. Can this value be propagated to existing buttons?

**Review by jverkoey — CHANGES_REQUESTED** [review]

File: components/ButtonBar/src/MDCButtonBar.h (hunk 7 above)
Comment (AlimoRabbani) [reply components/ButtonBar/src/MDCButtonBar.h]: Done, similar to updateButtonTitleColors in MDCButtonBar.

File: components/NavigationBar/src/MDCNavigationBar.h (hunk 6 above)
Comment (AlimoRabbani) [reply components/NavigationBar/src/MDCNavigationBar.h]: Done.

File: components/ButtonBar/src/MDCButtonBar.h (hunk 5 above)
Comment (AlimoRabbani) [reply components/ButtonBar/src/MDCButtonBar.h]: Done

**Comment by AlimoRabbani** [comment]
Made the changes requested.

**Review by jverkoey — APPROVED** [review]

**Comment by jverkoey** [comment]
Thank you! We'll merge once CI passes.

**Comment by AlimoRabbani** [comment]
Looks like Travis is timing out?

**Comment by jverkoey** [comment]
Looks like Travis did legitimately fail, but not because of your PR. You may want to rebase off of develop again.

**Comment by AlimoRabbani** [comment]
I'm going to rebase again and hope for the best. Looks like Travis logs exceeded 4MB and were cutoff. Not sure why it would exceed 4MB though.

**Comment by AlimoRabbani** [comment]
Looks like Travis is failing with the same error for other PRs as well :(

**Comment by jverkoey** [comment]
Aye. You can safely ignore the travis failures and the bazel/xcode9.2.0 appears to be a flake.

**Comment by AlimoRabbani** [comment]
Is there anything else I can do to help push this through?

**Pull request merged**